addons/web/static/tests/views/list_view_tests.js

```diff
@@ -259,6 +259,139 @@ QUnit.module("Views", (hooks) => {
         assert.isNotVisible(target.querySelector(".o_list_button_discard"));
     });
 
+    QUnit.test("select record range with shift click", async function (assert) {
+        await makeView({
+            type: "list",
+            resModel: "foo",
+            serverData,
+            arch: '<tree><field name="foo"/><field name="int_field"/></tree>',
+        });
+        await click(target.querySelectorAll(".o_data_row .o_list_record_selector input")[0]);
+        assert.containsOnce(target.querySelector(".o_cp_buttons"), ".o_list_selection_box");
+        assert.containsNone(target.querySelector(".o_list_selection_box"), ".o_list_select_domain");
```

Comment:
nitpicking: try to avoid unnecessary `.querySelector` thing
```suggestion
        assert.containsOnce(target, ".o_cp_buttons .o_list_selection_box");
        assert.containsNone(target, ".o_list_selection_box .o_list_select_domain");
```